src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -420,9 +441,21 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
 
     let needClean = false;
     if (changes) {
-      changes.forEachOperation((item, _, cindex) => {
-        if (item.previousIndex != null || (cindex < this._recordSize)) {
+      var lastRecord = this.lastRecord() + 1;
+
+      changes.forEachOperation((_, pindex, cindex) => {
+
+        // add new record after current position
+        if (pindex === null && (cindex < lastRecord)) {
+          console.debug('adding record after current position, no recalculation needed');
+          needClean = true;
```

Comment:
It might just be getting late for me, but shouldn't `needClean = true` be outside/after these two ifs since it should happen in all other cases?